verification/dependencies/github.com/google/gopacket/writer.gobra

```diff
@@ -17,9 +17,11 @@ type SerializableLayer interface {
 
 	requires  !opts.FixLengths
 	requires  b != nil && b.Mem()
+	requires  sl.Bytes(b.UBuf(), 0, len(b.UBuf()))
 	requires  Mem(ubuf)
 	preserves sl.Bytes(ubuf, 0, len(ubuf))
 	ensures   err == nil ==> (Mem(ubuf) && b.Mem())
+	ensures   err == nil ==> sl.Bytes(b.UBuf(), 0, len(b.UBuf()))
```

Comment:
I'm thinking about this spec and trying to puzzle this back together with the calls that may be performed to `p.buffer`. In particular, this spec implies that, after calling `SerializeTo`. if there is a failure, `p.buffer` may never be reused again. This does not seem to be compatible with the code of the router though, which always keeps the permissions to `p.Buffer.Mem()`